### ISSUE TITLE
Plotting using lonboard

### ISSUE THREAD

**Issue by benbovy** [benbovy/xdggs#11]
[lonboard](https://github.com/developmentseed/lonboard) is a new tool for efficient and interactive plotting of (vector) geospatial data in Jupyter.

It would be nice to leverage it for plotting DGGS data.

Lonboard uses deck.gl in the front-end. Although this is not (yet) available in lonboard, Deck.gl has specific layers for [H3](https://deck.gl/docs/api-reference/geo-layers/h3-cluster-layer) and [S2](https://deck.gl/docs/api-reference/geo-layers/s2-layer) so for those two grids this could potentially be *very* efficient as we would only need to transfer cell ids and cell values (colors) to the front-end and let deck.gl compute and render the cells.

For the other grids, we could fallback to transfer the grid cell geometries as polygons (see #10) and use lonboard's [SolidPolygonLayers](https://developmentseed.org/lonboard/latest/api/layers/solid-polygon-layer/).

cc @kylebarron

**Comment by kylebarron** [comment]
I think you probably want the [`H3HexagonLayer`](https://deck.gl/docs/api-reference/geo-layers/h3-hexagon-layer) instead of the clustering variant. The main annoyance was the h3 layer is that h3-js doesn't support the uint64 representation of an h3 index, so you either have to transmit the data to JS as strings or convert the uint64 to a hex string once you get there. Are s2 tokens always strings?

Either way, they're relatively layers easy to wrap, at least putting off h3 integers for later

**Comment by benbovy** [comment]
> The main annoyance with the h3 layer is that h3-js doesn't support the uint64 representation of an h3 index, so you either have to transmit the data to JS as strings or convert the uint64 to a hex string once you get there. Are s2 tokens always strings?

Yes I guess we'd have to do the same for converting s2 cell (uint64) ids to s2 tokens.

**Comment by kylebarron** [comment]
> The only issue I had is that I couldn't yet figure out how to zoom in on polygons in the range of 180° to 360° longitude, the map always snaps back to -180° to 0°, which does not have any data.

Not sure I fully understand, but maybe we need to enable map-repeating? https://deck.gl/docs/whats-new#world-repeating-in-web-mercator-maps

**Comment by keewis** [comment]
that could be it. Let me try to create a dummy dataset so you can see what I'm talking about.

**Comment by keewis** [comment]
If you want to reproduce, you'll need to have an environment with `xdggs` (installed from `main`). Something like this, if you use `mamba` / `conda`:
```yaml
channels:
- conda-forge
dependencies:
- python=3.12
- jupyterlab
- geopandas
- xarray
- numpy<2.1
- lonboard
- matplotlib
- pip
- pip:
  - git+https://github.com/xarray-contrib/xdggs.git
```
and here's the code:
```python
import geopandas as gpd
import lonboard
import numpy as np
import xarray as xr
import xdggs
from lonboard.colormap import apply_continuous_cmap
from matplotlib import colormaps
from matplotlib.colors import CenteredNorm, Colormap

resolution = 5

ds = (
    xr.Dataset(
        coords={
            "cell_ids": (
                "cells",
                np.arange(12 * 4**resolution),
                {
                    "grid_name": "healpix",
                    "resolution": resolution,
                    "indexing_scheme": "nested",
                },
            )
        }
    )
    .pipe(xdggs.decode)
    .pipe(lambda ds: ds.merge(ds.dggs.cell_centers()))
    .assign(data=lambda ds: np.sin(np.radians(ds["latitude"])))
    .assign_coords(cell_boundaries=lambda ds: ds.dggs.cell_boundaries())
)

gdf = ds.to_pandas().pipe(gpd.GeoDataFrame, geometry="cell_boundaries", crs=4326)

data = gdf["data"].to_numpy()
normalizer = CenteredNorm(vcenter=0, halfrange=1)
normalized_data = normalizer(data)

cmap = colormaps["coolwarm"]
colors = apply_continuous_cmap(normalized_data, cmap)

layer = lonboard.SolidPolygonLayer.from_geopandas(
    gdf, filled=True, get_fill_color=colors
)
map = lonboard.Map(layer)
map
```

**Comment by kylebarron** [comment]
Are you able to post a screenshot/screencast of the issue?

**Comment by keewis** [comment]
I can try to create a screencast, yes (I just need to figure out how to do that on my OS)

**Comment by keewis** [comment]
here it is:

![snapping-map](https://github.com/user-attachments/assets/b9b351a9-1125-4d7f-a799-a1168a6fe038)
